rbac/migration_tool/ingest.py

```diff
@@ -47,7 +53,16 @@ def extract_info_into_v1_role(role: Role):
         res_defs = [res_def for res_def in perm_res_defs.get((role_id, perm), [])]
         v1_perm = V1permission(perm_parts[0], perm_parts[1], perm_parts[2], frozenset(res_defs))
         v1_perms.append(v1_perm)
-    return V1role(role_id, frozenset(v1_perms), frozenset())  # we don't get groups from the sheet
+
+    # With the replicated role bindings algorithm, role bindings are scoped by group, so we need to add groups
+    # TODO: We don't need to care about principals here – see RHCLOUD-35039
+    # Maybe not even groups? See RHCLOUD-34786
+    groups = set()
+    for policy in role.policies.all():
+        principals = [str(principal) for principal in policy.group.principals.values_list("uuid", flat=True)]
+        groups.add(V1group(str(policy.group.uuid), frozenset(principals)))
+
+    return V1role(role_id, frozenset(v1_perms), frozenset(groups))  # we don't get groups from the sheet
```

Comment:
Can we drop this comment now that we've got the groups?

Reply:
Nice catch